crates/hotshot/types/src/epoch_membership.rs

```diff
@@ -247,8 +247,14 @@ fn spawn_catchup<T: NodeType>(coordinator: EpochMembershipCoordinator<T>, epoch:
             tx
         };
         // do catchup
-        let ret = coordinator.catchup(epoch).await;
-        let _ = tx.broadcast_direct(ret).await;
+
+        let result = coordinator.clone().catchup(epoch).await;
+        let _ = tx.broadcast_direct(result.clone()).await;
```

Comment:
Is the error handled properly on the receiving end?

Reply:
No it spawns catchup again
`
 let Ok(Ok(mem)) = rx.recv_direct().await else {
            return self.clone().catchup(epoch).await;
        };
        
        `
        
        I thought this was intentional, but maybe not?

Reply:
I thought this was intentional, but maybe not?

Reply:
Yeah I think it's fine, I added detail to the other comment